tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java

```diff
@@ -1535,6 +1535,46 @@ public void testTaggingNoLimit() throws Exception {
     testTagging(objectKey, ImmutableMap.copyOf(tagMap));
   }
 
+  @Test
+  public void testCopyObject() throws Exception {
+    final String bucketName = "bucket";
+    final String objectKey = "object";
+    final String targetObject = "/tmp/target";
```

Comment:
I would like to add a description to this test (comments) to explain what we're checking is that you can copy an object from one folder to a different folder, and those sub-directories will get created.

Reply:
added